### ISSUE TITLE
[JIT] Tracer throws runtime exception for torch.normal

### ISSUE THREAD

**Issue by neerajprad** [pytorch/pytorch#11504]
## Issue description

The normal distribution (and hence all distributions that use the `torch.normal` sampler) is throwing a runtime exception under JIT (this seems to be a recent regression). I have isolated the issue to the following code snippet which throws an exception on PyTorch master, and is breaking many PyTorch models. cc. @zou3519, @apaszke, @fritzo. 

## Code example

The following code that returns a sample from a normal distribution,

```python
def fn(mean, std):
    return torch.normal(mean, std)

compiled_fn = torch.jit.trace(fn, (torch.zeros(2, 3), torch.ones(2, 3)))
```

throws a Runtime Exception:
```
Traceback (most recent call last):
  File "/Users/npradhan/workspace/pyro_dev/pyro/examples/ex1.py", line 10, in <module>
    compiled_fn = torch.jit.trace(fn, (torch.zeros(2, 3), torch.ones(2, 3)))
  File "/Users/npradhan/miniconda2/envs/pytorch-master/lib/python3.6/site-packages/torch/jit/__init__.py", line 501, in trace
    module._create_method_from_trace('forward', func, example_inputs)
  File "/Users/npradhan/workspace/pyro_dev/pyro/examples/ex1.py", line 7, in fn
    return d.sample()
  File "/Users/npradhan/miniconda2/envs/pytorch-master/lib/python3.6/site-packages/torch/distributions/normal.py", line 54, in sample
    return torch.normal(self.loc.expand(shape), self.scale.expand(shape))
RuntimeError: Found an unsupported argument type in the JIT tracer: at::Generator*. File a bug report.
```

## System Info

```
  $ python collect_env.py
Collecting environment information...
PyTorch version: 0.5.0a0+f2f43ad
Is debug build: No
CUDA used to build PyTorch: None

OS: Mac OSX 10.13.3
GCC version: Could not collect
CMake version: version 3.12.0

Python version: 3.6
Is CUDA available: No
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA

Versions of relevant libraries:
[pip] numpy (1.15.0)
[pip] torch (0.5.0a0+f2f43ad)
[pip] torchfile (0.1.0)
[pip] torchvision (0.2.1)
[conda] torch                     0.5.0a0+f2f43ad           <pip>
[conda] torch                     0.5.0a0+2158f4a           <pip>
[conda] torchfile                 0.1.0                     <pip>
[conda] torchvision               0.2.1                     <pip>
```

**Comment by apaszke** [comment]
I'm fixing this right now.

**Comment by neerajprad** [comment]
@apaszke - Just FYI, it seems like this could be a bigger issue and I'm noticing this in `torch._standard_gamma` as well.

**Comment by osadj** [comment]
@apaszke This is still not fixed in PT 2.0. Can you please take a look?

`normal = torch.normal(0, 1, size=input_data.size(), generator=rng)
RuntimeError: Found an unsupported argument type in the JIT tracer: at::Generator. File a bug report.`
